packages/time-picker/src/vaadin-time-picker.js

```diff
@@ -471,21 +519,37 @@ class TimePicker extends PatternMixin(InputControlMixin(ThemableMixin(ElementMix
     this._comboBoxValue = this.i18n.formatTime(objWithStep);
     this.__useMemo = false;
 
-    this.validate();
-    this.__commitPendingValue();
+    this.__commitValueChange();
   }
 
-  /** @private */
-  __commitPendingValue() {
+  /**
+   * Depending on the nature of the value change that has occurred since
+   * the last commit attempt, triggers validation and fires an event:
+   *
+   * Value change             | Event
+   * -------------------------|-------------------
+   * empty => parsable        | change
+   * empty => unparsable      | unparsable-change
+   * parsable => empty        | change
+   * parsable => parsable     | change
+   * parsable => unparsable   | change
+   * unparsable => empty      | unparsable-change
+   * unparsable => parsable   | change
+   * unparsable => unparsable | unparsable-change
+   *
+   * @private
+   */
+  __commitValueChange() {
     if (this.__committedValue !== this.value) {
-      this.__dispatchChange();
-      this.__committedValue = this.value;
+      this.validate();
+      this.dispatchEvent(new CustomEvent('change', { bubbles: true }));
+    } else if (this.__committedUnparsableValue !== this.__unparsableValue) {
+      this.validate();
+      this.dispatchEvent(new CustomEvent('unparsable-change'));
```

Comment:
Should this not bubble (like "change")?

Reply:
I had similar thoughts about it. None of the other custom events we use bubble, but then again, listening to both, "change" and "unparsable-change", on a parent wouldn't be doable if "unparsable-change" didn't bubble. Maybe we should discuss this tomorrow with the team.